strapi/strapi
# [TS] Adds types for the publicationState param

### What does it do?

- Add type utility for the { publicationState } (params)
  - Only present in the param object if the given schema uid has publication state enabled
  - Can be used with or without generated types, if types are not generated, the param will always be there for content types

### Why is it needed?

Part of the params typing initiative

### How to test it?

- Head to packages/core/strapi/lib/services/entity-service/types/params/index.d.ts and rename the file to index.ts
- Adds the following code snippet
  ```ts
  const params: PublicationState.For<Common.UID.Schema> = {
    // ...
  }
  ```
- Since types are not generated the object should contain the `publicationState` optional param
- Then, generate types for the getstarted example project
- Add those lines to the top of the file you're in
  ```ts
  import '../../../../../../../../examples/getstarted/types/generated/contentTypes'
  import '../../../../../../../../examples/getstarted/types/generated/components'
  ```
- Modify the declaration of your params by replacing Common.UID.Schema by an actual UID from a schema that has been generated
- Play around with the object, this time the object should have the publicationState param or not depending on the content-type configuration
- You can also test with components to make sure the param is not here
